pkg/xds/generator/admin_proxy_generator.go

```diff
@@ -41,13 +63,39 @@ func (g AdminProxyGenerator) Generate(ctx xds_context.Context, proxy *core_xds.P
 
 	resources := core_xds.NewResourceSet()
 
+	for _, se := range staticEnpointPaths {
+		se.ClusterName = envoyAdminClusterName
+	}
+
+	for _, se := range staticTlsEnpointPaths {
+		se.ClusterName = envoyAdminClusterName
+
+		token, err := ctx.EnvoyAdminClient.GenerateAPIToken(proxy.Dataplane)
+		if err != nil {
+			return nil, err
+		}
+		se.Header = "Authorization"
+		se.HeaderExactMatch = fmt.Sprintf("Bearer %s", token)
+	}
+
+	keyPair, err := NewSelfSignedCert("admin", tls.ServerCertType, "localhost")
+	if err != nil {
+		return nil, err
+	}
+
 	// We bind admin to 127.0.0.1 by default, creating another listener with same address and port will result in error.
 	if proxy.Dataplane.Spec.GetNetworking().Address != "127.0.0.1" {
 		listener, err := envoy_listeners.NewListenerBuilder(proxy.APIVersion).
 			Configure(envoy_listeners.InboundListener(envoy_names.GetAdminListenerName(), proxy.Dataplane.Spec.GetNetworking().Address, adminPort)).
-			Configure(envoy_listeners.FilterChain(envoy_listeners.NewFilterChainBuilder(proxy.APIVersion).
-				Configure(envoy_listeners.StaticEndpoint(envoy_names.GetAdminListenerName(), adminReadyEndpoint, adminReadyEndpoint, envoyAdminClusterName)),
-			)).
+			Configure(envoy_listeners.TLSInspector()).
+			Configure(
+				envoy_listeners.FilterChain(envoy_listeners.NewFilterChainBuilder(proxy.APIVersion).
+					Configure(envoy_listeners.StaticEndpoints(envoy_names.GetAdminListenerName(), staticEnpointPaths)),
+				),
+				envoy_listeners.FilterChain(envoy_listeners.NewFilterChainBuilder(proxy.APIVersion).
+					Configure(envoy_listeners.FilterChainMatch("tls")).
+					Configure(envoy_listeners.StaticTlsEndpoints(envoy_names.GetAdminListenerName(), &keyPair, staticTlsEnpointPaths)),
+				)).
```

Comment:
nice, that look pretty awesome that we can easily compose many filter chains with matchers.